textcursor.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2015 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gxui
+
+import "github.com/google/gxui/interval"
+
+// TextCursor represents a text position and selection. A selection is defined as
+// the pair of cursor indices sorted least to greatest. When transforming a
+// cursor with no selection, mark should equal index.
+type TextCursor struct {
+	index int
```

Comment:
I've considered this as well but I'm not sure what the ramifications are for redrawing depending on someones expectations, whether pushing new controls to gxui or writing an app against gxui that does something novel.

Ultimately what I had in mind was to be able to retrieve an instance of TextCursor from a controller for read-only stuff, providing a more concrete type from method returns and replacing methods like `TextBoxController.Selection(int) int` `TextBoxController.Caret(int) int` and all those other `(int) int` methods with a smaller set that does the same thing.

I'm pretty sure this would result in a simpler api for TextBoxController but I haven't finished thinking it all the way through yet.